internal/graph/check.go

```diff
@@ -444,3 +463,38 @@ func combineResponseMetadata(existing *v1.ResponseMeta, responseMetadata *v1.Res
 	combined.DebugInfo = debugInfo
 	return combined
 }
+
+func dispatchAllAsync(
+	ctx context.Context,
+	requests []ReduceableCheckFunc,
+	resultChan chan<- CheckResult,
+	concurrencyLimit uint16,
+) func() {
+	sem := make(chan struct{}, concurrencyLimit)
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+	dispatcher:
```

Comment:
Why the decision to use this vs the new limiting errorgroup?

Reply:
The errgroup approach was causing too much churn to the interfaces. We can port over to that one by one, but coupling sweeping interface changes was making that PR more risky than it had to be.